SendGrid/SendGrid/Client.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Text;
+using SendGrid.Resources;
+using System.Web.Script.Serialization;
+using System.Net;
+
+namespace SendGrid
+{
+    public class Client
+    {
+        private string _apiKey;
+        public APIKeys ApiKeys;
+        private Uri _baseUri;
+
+        /// <summary>
+        ///     Create a client that connects to the SendGrid Web API
+        /// </summary>
+        /// <param name="apiKey">Your SendGrid API Key</param>
+        /// <param name="baseUri">Base SendGrid API Uri</param>
+        public Client(string apiKey, string baseUri = "https://api.sendgrid.com/")
+        {
+            _baseUri = new Uri(baseUri);
+            _apiKey = apiKey;
+            ApiKeys = new APIKeys(this);
+        }
+
+        /// <summary>
+        ///     Create a client that connects to the SendGrid Web API
+        /// </summary>
+        /// <param name="method">HTTP verb, case-insensitive</param>
+        /// <param name="endpoint">Resource endpoint, do not prepend slash</param>
+        /// <param name="data">An object representing the resource's data</param>
+        /// <returns>An asyncronous task</returns>
+        private async Task<HttpResponseMessage> RequestAsync(string method, string endpoint, Object data)
```

Comment:
Additionally, perhaps passing in a Uri type so you are guaranteed earlier in the process to be getting a valid url.